src/main/java/com/aquaticinformatics/aquarius/sdk/helpers/SdkServiceClient.java

```diff
@@ -235,6 +235,30 @@ public void setBaseUrl(String baseUrl){
         return propertyMap;
     }
 
+    protected String removeUnescapedQuotes(String source, int startIndex) {
```

Comment:
Hi Zachary. I'm finally coming up for air after my vacation, so it's time to review this pull request.

Your solution is on the right track (removing the unescaped quotes), but I'd like to avoid the recursive calls contained within this method.

Since I'm a fan of [regular expressions](https://xkcd.com/208/), I think this could be simplified into a single flow-of-control method like this:

```java
import java.util.regex.Pattern;
import java.util.regex.Matcher;

private static Pattern unescapedQuotePattern;

static {
    unescapedQuotePattern = Pattern.compile("([^\\\\])(\"))"); // A quote not preceeded by a backslash
}

private String removeUnescapedQuotes(String source) {
    return unescapedQuotePattern
        .matcher(source)
        .replaceAll("$1"); // Keep whatever preceeded the unescaped quote
}
```

Can you rework your PR and tests to use this approach?

Reply:
I originally attempted to use a regex: `(?<!\\)(?:\\{2})*\K"` but couldn't find a good way to replicate the \K behavior since Java's regex engine doesn't support that switch. I think I might have an idea on how to make it work now though.